app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb

```diff
@@ -193,6 +165,27 @@ def initialize_vm(vm_hash)
       def create_managed_virtual_machine(vm_hash)
         vm_params = initialize_vm(vm_hash)
         vm_params.network_profile = define_network_profile(vm_hash[:network_interface_card_ids])
+        # If image UUID begins with / it is a custom managed image
+        # Otherwise it is a marketplace URN
+        unless vm_hash[:vhd_path].start_with?('/')
+          urn = vm_hash[:vhd_path].split(':')
+          publisher = urn[0]
+          offer     = urn[1]
+          sku       = urn[2]
+          version   = urn[3]
+          vm_hash[:vhd_path] = nil
+        end
+
+        if vm_hash[:vhd_path].nil?
+          # We are using a marketplace image
+          vm_params.storage_profile.image_reference = image_reference(publisher, offer,
+                                                            sku, version)
+        else
+          # We are using a custom managed image
+          image_ref = ComputeModels::ImageReference.new
+          image_ref.id = vhd_path
+          vm_params.storage_profile.image_reference = image_ref
+        end
```

Comment:
Can we put this part into a separate method, as we do for `define_network_profile`?